http/api_redirect.go

```diff
@@ -0,0 +1,27 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/hashicorp/vault/vault"
+)
+
+func handleApiRedirect(core *vault.Core, base http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		redir, err := core.GetAPIRedirect(r.Context(), r.URL.Path)
+		if err != nil {
+			core.Logger().Warn("error resolving potential API redirect", "error", err)
+		} else {
+			if redir != "" {
+				w.Header().Set("Location", redir)
+				w.WriteHeader(http.StatusFound)
```

Comment:
Should we make this logic redirect 302 on GET/HEAD requests only and use 307 for other request types?

I've been burned by bad browser behavior [in the past around 302 handling](https://en.wikipedia.org/wiki/HTTP_302)...

Reply:
TiL